src/sidebar/store/modules/viewer.js

```diff
@@ -11,6 +15,14 @@ const initialState = {
    * current state of the sidebar, but tracks whether it has ever been open
    */
   sidebarHasOpened: false,
+
+  /**
+   * Data for the content information banner shown above the content in the main
+   * guest frame.
+   *
+   * @type {ContentInfoConfig|null}
+   */
+  contentInfo: null,
```

Comment:
This is not the most obvious place for this info to live. But it doesn't neatly fit into any of the other modules. Perhaps we should rename `viewer.js` to something else (`other.js`?) to reflect its status as a home for isolated data that doesn't fit into an existing category.

Reply:
`sidebarHasOpened` feels "correct" for a module named `viewer.js`. I grok it as "this is store state related to the sidebar as a viewer or container".

I'm with you that `contentInfo` feels awkward here. Of the modules we currently have—if I had to choose from existing ones—it seems most associated with `frames`. `contentInfo` is further awkward in that it seems like something that would be associated with a given document (or, by extension, a frame with a document), but it simultaneously applies to the entire "session" or "launch." Gives one pause, eh?

Unless we redesign things, there's a maximum of one embedder frame that can be sending (or receiving) messages to (or from) the sidebar using the current RPC mechanism for that purpose. That means we'll only ever get "content info" data from one source...Perhaps one way to look at this is that if there is an embedder frame it is assumed there is only one guest? Or is that too facile?

Is having a store module for `embedder` or similar plausible? I dunno, spitballing.

Reply:
> I'm with you that contentInfo feels awkward here. Of the modules we currently have—if I had to choose from existing ones—it seems most associated with frames. contentInfo is further awkward in that it seems like something that would be associated with a given document (or, by extension, a frame with a document), but it simultaneously applies to the entire "session" or "launch." Gives one pause, eh?

I agree that `frames` is probably a more appropriate place for this. Some refactoring of the frames module is required to move it there though. See https://github.com/hypothesis/client/pull/4682.

> contentInfo is further awkward in that it seems like something that would be associated with a given document (or, by extension, a frame with a document), 

You're right. For the current and foreseeable use cases though, it simplifies the design if we can assume that scenarios that use the banner only need to show it in the "main" guest. This avoids issues such as how the LMS indicates which guest the content info should be sent to (by URL? by PDF fingerprint? etc.) I figure we can solve this problem when the need arises.

Reply:
> Is having a store module for embedder or similar plausible? I dunno, spitballing.

Sure, if we have a collection of fields that are clearly embedder related. I can imagine that we _might_ want to use the content banner outside of the LMS in future, eg. if Hypothesis became a feature on the main JSTOR website or something, but there are no concrete plans yet.